dev-docs/007-transforms-rfc.md

```diff
@@ -0,0 +1,228 @@
+
+### COMPLETE and INCREMENTAL
+
+These modes fundamentally deal with creating and adding data that is subject to the TIME_WINDOW
+configuration. The intention is that dynamic poses and links that are updated frequently, defined as
+being within the TIME_WINOW, would be sent in either of these messages.
+
+The semantics of _INCREMENTAL_ are at the stream set level, so it is worth noting that transforms
+are not combined but an _INCREMENTAL_ message would simply replace the active pose at that
+timestamp.
+
+Links should be avoided in these messages as a situation where the link would be only shortly lived
+is unclear in the current use-cases. They can be defined but will only be present for the duration
+of TIME_WINDOW.
+
+Keep in mind that these messages may also be purged as the assumption in an XVIZ client is that a
+minimal amount of state is necessary.
+
+### PERSISTENT
+
+PERSISTENT messages is where most links should be defined as they setup a structure that changes
+very little if at all. Static poses for a link, like those from a platform to a sensor, should also
+be defined in this message type.
+
+PERSISTENT messages define state that is not subject to the TIME_WINDOW and also less likely to be
+purged. The messages are still subject to a timestamp and therefore care must be taken to ensure
+this data does not grow unbounded in the application state.
+
+## Streetscape.gl implementation details
+
+The streetscape.gl implementation currently manages the transform resolution through a single
+function, resolveCoordinateTransform(), which is given access to the the internal data frame and the
+the stream metadata. With that information it can process the _links_ transform chain to compute the
+affective matrix for rendering.
+
+We may want to build a transform cache if necessary, but that is a performance implmentation detail.
+
+# Follow-up proposals
+
+The following are proposals that are related to this but are not required for this proposal to be
+accepted. They are mentioned only to fully understand the scope, influence, and considerations that
+have helped contextualize and shape this proposal.
+
+- Making timestamps optional on the streamset
+- Support for linear interpolation on poses
+
+# Other considerations
+
+### Metadata only approach
+
+An attempt was made to fit this into stream metadata, and make metadata additions more dynamic.
+However the criteria that the links could also be time relative turns this into _data_ rather than
+metadata.
+
+Further complications with this approach would be an addition that allows incremental metadata
+definitions. In _on-line_ data cases the streams being sent may not be known until they are first
+encountered. This makes requiring all stream metadata, easy in an off-line situation, pratically
+impossible.
+
+### Rendering consideration
+
+In deck.gl, used by streetscape.gl, the layers are separated by geometry type and we render all data
```

Comment:
Maybe call out here that the current link design does not allow for per object transforms at all here?

Reply:
ok